src/ecs/systems/rendering.rs

```diff
@@ -15,25 +21,26 @@ pub struct RenderSystem {
     #[derivative(Debug = "ignore")]
     renderer: Renderer,
     scene: Scene,
+    #[derivative(Debug = "ignore")]
+    factory: Arc<Factory>,
```

Comment:
We should just clone the `Factory`. If I didn't do that already, the queue contained in it should be reference counted.